packages/sw-lib/src/lib/sw-lib.js

```diff
@@ -351,6 +352,25 @@ class SWLib {
       requestWrapper: new RequestWrapper(wrapperOptions),
     });
   }
+
+  /**
+   * This method will register any default routes the library will need.
+   */
+  _registerDefaultRoutes() {
+    const revisionedManager = this._precacheManager.getRevisionedCacheManager();
+    const cacheFirstHandler = this.cacheFirst({
+      cacheName: revisionedManager.getCacheName(),
+    });
+
+    const route = new this.Route({
+      match: ({url, event}) => {
+        const cachedUrls = revisionedManager.getCachedUrls();
```

Comment:
Yeah, there are a couple of places in the various modules where we assume that properties will never change, even though they're not actually protected via setters or immutable or anything like that.

I've kind of just been hoping that it won't be a problem, since developers would normally have to go out of their way to explicitly change the internal state. We should revisit if we come across API surfaces that developers might accidentally use and which have those potential side effects.